spec/models/document_spec.rb

```diff
@@ -2,31 +2,74 @@
-              Lesson.new(id: 8, title: "test lesson1", description: "test lesson1", created_at: "2021-05-19 21:44:01", updated_at: "2021-05-19 21:44:01", units_covered: "3", course_id: 4)
-              User.new(id: 1, email: "adi_111@icloud.com", role: "teacher", created_at: "2021-05-05 07:06:27", updated_at: "2021-05-05 07:08:29", password: "Pa$$word1")}
+    before do
+      @user = FactoryBot.create(:user)
+      @tag = FactoryBot.create(:tag)
+      @course = FactoryBot.create(:course)
+      lesson_hash = {title: "test lesson1", description: "test lesson1", 
+        created_at: Time.now, updated_at: Time.now, units_covered: "3", course_id: @course.id}
+      @lesson = Lesson.create(lesson_hash)
+      document_hash = {name: "Doc1 for RonR", description: "Doc1 for RonR", kind: "type1", created_at: "2021-06-01 03:25:33", updated_at: "2021-06-01 03:25:33", lesson_id: @lesson.id, course_id: @course.id}
+      @document = Document.create(document_hash) 
+    end
+
+    after do
+      @user.delete
+      @tag.delete
+      KeyWord.all.delete_all
+      @document.delete
+      @lesson.delete
+      @course.delete
+    end
+    
     it "is valid with valid attributes" do
-      expect(subject).to be_valid    
+      expect(@document).to be_valid  
+    end
+    it "is not valid without name" do
+      @document.name = nil
+      expect(@document).to_not be_valid
+    end
+    it "is not valid without description" do
+      @document.description = nil
+      expect(@document).to_not be_valid
+    end
+    it "is not valid without kind" do
+      @document.kind = nil
+      expect(@document).to_not be_valid
     end
   end
   describe "Associations" do
-    it { should belong_to(:course).optional }
-    it { should belong_to(:lesson).optional }
-    # it { should have_one(:document) }
- end
-  it "is not valid without a name" do
-    subject.name = nil
-    expect(subject).to_not be_valid
-  end
-  it "is not valid without a description" do
-    subject.description = nil
-    expect(subject).to_not be_valid
-  end
-  it "is not valid without a kind" do
-    subject.kind = nil
-    expect(subject).to_not be_valid
-  end
-end
+    before do
+      @user = FactoryBot.create(:user)
+      @tag = FactoryBot.create(:tag)
+      @course = FactoryBot.create(:course)
+      lesson_hash = {title: "test lesson1", description: "test lesson1", 
+        created_at: Time.now, updated_at: Time.now, units_covered: "3", course_id: @course.id}
+      @lesson = Lesson.create(lesson_hash)
+      document_hash = {name: "Doc1 for RonR", description: "Doc1 for RonR", kind: "type1", created_at: "2021-06-01 03:25:33", updated_at: "2021-06-01 03:25:33", lesson_id: @lesson.id, course_id: @course.id}
```

Comment:
Just use the lesson_id in this hash.  We can check course_id in the course test.

Reply:
Chuck I used the course hash because I checked in the DB the Document.all when we added a document for courses. Also in the DB for this table we have a course_id.=> #<ActiveRecord::Relation [#<Document id: 3, name: "Doc course RonR", description: "Doc course RonR", kind: "type1", created_at: "2021-06-02 23:36:30", updated_at: "2021-06-02 23:36:30", lesson_id: nil, course_id: 6>]>